Nonprofit-Exchange-Hub/web-app
# Asset images #212

#212  

Dev Summary
Added a feature to enable the user to upload image urls when filling the Goods Needed form. Includes url validation and uniqueness of the url.

An array containing 1 empty string is the default state from which this portion of the form is rendered. For each element in the array, a text input field is rendered. Clicking a button to add another photo url appends 1 additional empty string to the array, thereby rendering an additional input field. The user should add one image url to each field. If the user enters a string that is not a valid url, an error message is displayed by <FormHelperText>.

Yup validation schemas were created in a separate file, NeedFormGoods-validation.ts. urlSchema is used to check each image url is a valid url. I included a custom method to check if each url is unique that runs on the validationSchema when the form is submitted, but this needs to have additional work done to display an error to the user if we want to keep this check.


Test Plan
1. Entered many valid and invalid urls into the input fields
2. Attempted entering duplicate urls
3. Confirmed the form submission successfully posted images to the backend

repro steps:

1. http://localhost:3000/need/goods
2. Scroll to very bottom of the form to see input fields below "Or link photos below"
3. Add first image url to input field labeled "Photo 1"
4. Click button "Click here to add another photo"
5. Repeat the process for the desired number of images to upload.
6. Fill out the rest of the form and submit it.
expected: http://localhost:3001/api/assets -- latest object includes "imgUrls" property with an array of the urls entered on the frontend.


![Screen Shot 2022-10-31 at 9 08 59 PM](https://user-images.githubusercontent.com/101307197/199137279-548ec6f9-474d-4e38-b51f-3dc745ec7353.png)

![Screen Shot 2022-10-31 at 9 11 39 PM](https://user-images.githubusercontent.com/101307197/199137286-50639748-591f-407e-8575-d11429797bf2.png)
